### ISSUE TITLE
Cache strategy error

### ISSUE THREAD

**Issue by jrodriguezva** [PaNaVTEC/Clean-Contacts#18]
When you call the method "Contact obtain(String md5)", you return null if the cache is invalid. This will throw an exception if you wait a while to navigate the MainActivity to DetailActivity, since there would be no data to display in the DetailActivity. I apologize for my basic level of English, I just speak Spanish.

**Comment by PaNaVTEC** [comment]
That is completely true, the aim of the repository was to demonstrate how to architect android and I was not focused and that kind of edge cases but yes, is a bug and must be solved
